### PR TITLE
Code cleanup and Refactors by Stefan Monnier

### PR DESCRIPTION
**Pull request by MatthewZMD**
PR representing Stefan Monnier whom sent me the patch through email

- Prefer derived-mode-p to testing major-mode.
- Quote some vars in docstrings so they appear as hyperlinks in *Help*.
- Remove single-backslashes from strings when they don't do anything
  (e.g. `\/\/` is the same string as `//`).
- A few eta-like rewrites to reduce some code duplication.

### CODE REVIEW

File: eaf.el [hunk 1]
```diff
@@ -591,35 +589,33 @@ Use it as (eaf-bind-key var key eaf-app-keybinding)"
                        (h (nth 3 window-allocation))
                        )
                   (when (and
-                         (> mouse-press-x x)
-                         (< mouse-press-x (+ x w))
-                         (> mouse-press-y y)
-                         (< mouse-press-y (+ y h)))
+                         (< x mouse-press-x (+ x w))
+                         (< y mouse-press-y (+ y h)))
```
Comment (manateelazycat) [inline eaf.el]: Why remove ```(> mouse-press-x x)``` and ```(> mouse-press-y y)``` ?

File: eaf.el [hunk 2]
```diff
@@ -549,7 +547,7 @@ Please ONLY use `eaf-bind-key' to edit EAF keybindings!"
         ;; Send key to Python side if key-command is single character key.
         ((or (equal key-command "self-insert-command")
              (equal key-command "completion-select-if-within-overlay")
-             (equal key-command "nil")
+             (eq key-command 'nil)
```
Comment (manateelazycat) [inline eaf.el]: symbol-name always return "nil" string, 
Have you test this change?

Comment (MatthewZMD) [reply eaf.el]: I tested it and don't see any problem. `eq` compares the object `nil` and `key-command` right? If you feel unsafe, I can change it back.

File: eaf.el (hunk 1 above)
Comment (MatthewZMD) [reply eaf.el]: `(and (> mouse-press-y y) (< mouse-press-y (+ y h)))` is the same as `(< x mouse-press-x (+ x w))`

File: eaf.el (hunk 2 above)
Comment (manateelazycat) [reply eaf.el]: I'm sure symbol-name always return string (eq "nil" 'nil) is nil , it's wrong

File: eaf.el (hunk 1 above)
Comment (manateelazycat) [reply eaf.el]: Correctly logic is:

                         (> mouse-press-x x)
                         (< mouse-press-x (+ x w))
                         (> mouse-press-y y)
                         (< mouse-press-y (+ y h))

Comment (manateelazycat) [reply eaf.el]: Oh, I see.

< support many params

**Comment by manateelazycat** [comment]
@MatthewZMD I push new https://github.com/manateelazycat/emacs-application-framework/commit/6b63a92594883b166d09bca615273b095bb921b6 fix eq problem.

**Pull request merged**